Darwin/Services/StitchRemoteMongoDBService/StitchRemoteMongoDBServiceTests/SyncIntTests.swift

```diff
@@ -2064,6 +2108,101 @@ class SyncIntTests: BaseStitchIntTestCocoaTouch {
         coll.verifyUndoCollectionEmpty()
     }
 
+    func testMultiUserSupport() throws {
+        let (remoteColl, coll) = ctx.remoteCollAndSync
+
+        var docToInsertUser1 = ["hello": "world"] as Document
+        var docToInsertUser2 = ["hola": "mundo"] as Document
+        var docToInsertUser3 = ["hallo": "welt"] as Document
+
+        // configure Sync
+        coll.configure(conflictHandler: failingConflictHandler)
+
+        let doc1Id = coll.insertOne(&docToInsertUser1)!.insertedId!
+        let doc1Filter = ["_id": doc1Id] as Document
+
+        // assert that the resolution is reflected locally, but not yet remotely
+        var expectedDocument = docToInsertUser1
+        XCTAssertEqual(expectedDocument.sorted(), coll.findOne(doc1Filter)!.sorted())
```

Comment:
Isnt this call to findOne async and needs to be waited on?

Reply:
check out the `private extension RemoteMongoCollection` in this file. We added versions of all the CRUD methods that return synchronously

Reply:
Oh wow, i see. I just pushed a new change to my findOne() that changes the internal findOne() in SyncIntTests.